pyprof/prof/index_slice_join_mutate.py

```diff
@@ -139,11 +139,7 @@ def __init__(self, d):
         args = list(filter(lambda x: x['name'] != 'out', args))
         assert (len(args) == 3)
 
-        #Get input
-        if (args[0]['name'] == ""):
-            arg = args[0]
-        else:
-            arg = list(filter(lambda x: x['name'] == "input", args))[0]
+        arg = args[0]
```

Comment:
For a function `foo(a, b, c)`, a user can call it as `foo(c=20, b=10, a=5)`. If we want `a` we cannot assume the arguments will appear in the order of the function signature and always pick the first one.

Reply:
same as above the arg filtering is broken - what are you trying to capture in the filter command?

Reply:
Let me explain with an example. Consider the pytorch API for `relu`

```
torch.nn.functional.relu(input, inplace=False)
```

It can be called in a variety of ways e.g.
```
a = torch.ones(5,5)
relu(a)
relu(a, inplace=True)
relu(input=a, inplace=True)
relu(inplace=True, input=a)
```

The purpose of filtering is to determine `a` in all cases. Depending on how the function was called it can be part of `args` or `kwargs`. That's the intent, and its possible in a few cases the code is off. There is an assertion after the argument capture to verify if we got it right. In the past 2 years, they have never fired (or barely).